fission/src/systems/simulation/synthesis_brain/SynthesisBrain.ts

```diff
@@ -42,13 +50,24 @@ class SynthesisBrain extends Brain {
 
         // Only adds controls to mechanisms that are controllable (ignores fields)
         if (mechanism.controllable) {
-            this.configureArcadeDriveBehavior();
-            this.configureArmBehaviors();
-            this.configureElevatorBehaviors();
-            this.configureInputs();
-
-            SynthesisBrain._currentRobotIndex++;
+            if (SynthesisBrain.numberRobotsSpawned[assemblyName] == undefined)
+               SynthesisBrain.numberRobotsSpawned[assemblyName] = 0 
+            else
+                SynthesisBrain.numberRobotsSpawned[assemblyName]++
+
+            this._assemblyIndex = SynthesisBrain.numberRobotsSpawned[assemblyName]
+            SynthesisBrain.robotsSpawned.push(`[${this._assemblyIndex}] ${assemblyName}`)
```

Comment:
It would probably be best if this could go in one place (like a function to create the name or something) that's used here and later on line 84 so that it's only specified in one place.